src/Common/EnvConfig/Client/ConfigProfile.php

```diff
@@ -0,0 +1,227 @@
+        public readonly ?ConfigCodec $codecConfig = null,
+    ) {
+        // Normalize empty strings to null
+        $this->address = $address === '' ? null : $address;
+        $this->namespace = $namespace === '' ? null : $namespace;
+        $this->apiKey = $apiKey === '' ? null : $apiKey;
+
+        // Normalize gRPC metadata keys to lowercase per gRPC spec
+        $meta = [];
+        foreach ($grpcMeta as $key => $value) {
+            $meta[\strtolower($key)] = \is_array($value) ? $value : [$value];
+        }
+        $this->grpcMeta = $meta;
+
+        // Validate codec is not configured (not supported in PHP SDK)
+        $codecConfig?->endpoint === null && $codecConfig?->auth === null or throw new CodecNotSupportedException();
+    }
+
+    /**
+     * Merge this profile with another profile, with the other profile's values taking precedence.
+     *
+     * Creates a new profile by combining settings from both profiles. Non-null values from the
+     * provided config override values from this profile. TLS and codec configurations are deeply merged.
+     * gRPC metadata arrays are merged with keys normalized to lowercase (per gRPC spec), with
+     * the other profile's values replacing this profile's values for duplicate keys.
+     *
+     * @param self $config Profile to merge with (values from this take precedence)
+     * @return self New merged profile
+     */
+    public function mergeWith(self $config): self
+    {
+        return new self(
+            address: $config->address ?? $this->address,
+            namespace: $config->namespace ?? $this->namespace,
+            apiKey: $config->apiKey ?? $this->apiKey,
+            tlsConfig: self::mergeTlsConfigs($this->tlsConfig, $config->tlsConfig),
+            grpcMeta: self::mergeGrpcMeta($this->grpcMeta, $config->grpcMeta),
+            codecConfig: self::mergeCodecConfigs($this->codecConfig, $config->codecConfig),
+        );
+    }
+
+    /**
+     * Convert this profile to ClientOptions.
+     *
+     * Creates a ClientOptions instance with the namespace from this profile.
+     * Other ClientOptions properties (identity, queryRejectionCondition) use their default values.
+     *
+     * @return ClientOptions Configured client options
+     */
+    public function toClientOptions(): ClientOptions
+    {
+        $options = new ClientOptions();
+        $this->namespace === null or $options = $options->withNamespace($this->namespace);
+
+        return $options;
+    }
+
+    /**
+     * Convert this profile to a configured ServiceClient.
+     *
+     * Creates a ServiceClient with proper TLS configuration and API key authentication.
+     * If tlsConfig is present and not disabled, creates an SSL-enabled client with optional
+     * mTLS support. If apiKey is present, configures authentication headers.
+     *
+     * @return ServiceClient Configured service client ready for use
+     * @throws \InvalidArgumentException If address is not configured
+     */
```

Comment:
We should enable TLS when the API key is provided, this is a special case.
Does `withAuthKey` achieve this even with `this->tlsConfig` is null?